[BUG] wrong Block.Header.LastResultsHash

**Describe the bug**
We ran a snapshot node on version Terra 0.5.9.  This snapshot was working fine on Terra 0.5.9 - 0.5.11 cluster nodes.
After the Mainnet network upgrade on Friday, December, 17 all nodes have stopped working with this error:

```
terrad[1376]: #033[90m3:20PM#033[0m #033[32mINF#033[0m ABCI Replay Blocks #033[36mappHeight=#033[0m5684022 #033[36mmodule=#033[0mconsensus #033[36mstateHeight=#033[0m5684022 #033[36mstoreHeight=#033[0m5684023
terrad[1376]: #033[90m3:20PM#033[0m #033[32mINF#033[0m Replay last block using real app #033[36mmodule=#033[0mconsensus
terrad[1376]: Error: error during handshake: error on replay: wrong Block.Header.LastResultsHash.  Expected C2743948675AB3AAF1709585B5DE7425EC978660F209276E0A46ABB8A53C9DED, got 4368B26106D5B1171A324A42021D73DA2C814A87BDB39BCC68863E7418F29325
2
```
Since that moment no versions able to sync up with the old snapshot. The last snapshot version from December, 16 was 100% working at the moment of creation.

**Expected behavior**
Old file snapshot should be converted by a new application version without failures.

**To Reproduce**
Run file snapshot with the date of December, 16 on the latest version 0.5.13

**(if applicable) suggested solution**
We are resyncing all nodes from scratch which is an extremely long procedure.

I built a statically linked library based on the Docker receipt.

Download `/lib/libwasmvm_muslc.a` then run
```
LEDGER_ENABLED=false BUILD_TAGS=muslc make build
```

With snapshot working on 16 December before the soft-fork got the same error:

```
panic: Failed to process committed block (5684023:1EB882EF91752B97EA1AB8D29D0CBABC5CD5C27D2516CED0F0E638F507DDD6E9):
wrong Block.Header.LastResultsHash.
Expected C2743948675AB3AAF1709585B5DE7425EC978660F209276E0A46ABB8A53C9DED,
got 4368B26106D5B1171A324A42021D73DA2C814A87BDB39BCC68863E7418F29325```

Here is the detail of the build. Not sure how to check if libwasm version is correct.

```
# terrad version --long
name: terra
server_name: terrad
version: 0.5.13
commit: 0909bd43523fe9a77009fc4526e093b5bb0fecd8
build_tags: netgo,muslc
go: go version go1.17.5 linux/amd64
build_deps:
- filippo.io/edwards25519@v1.0.0-beta.2
- github.com/99designs/keyring@v1.1.6 => github.com/cosmos/keyring@v1.1.7-0.20210622111912-ef00f8ac3d76
- github.com/ChainSafe/go-schnorrkel@v0.0.0-20200405005733-88cbf1b4c40d
- github.com/CosmWasm/wasmvm@v0.16.2
- github.com/Workiva/go-datastructures@v1.0.52
- github.com/armon/go-metrics@v0.3.9
- github.com/beorn7/perks@v1.0.1
- github.com/bgentry/speakeasy@v0.1.0
- github.com/btcsuite/btcd@v0.22.0-beta
- github.com/cespare/xxhash/v2@v2.1.1
- github.com/coinbase/rosetta-sdk-go@v0.6.10
- github.com/confio/ics23/go@v0.6.6
- github.com/cosmos/btcutil@v1.0.4
- github.com/cosmos/cosmos-sdk@v0.44.5
- github.com/cosmos/go-bip39@v1.0.0
- github.com/cosmos/iavl@v0.17.3
- github.com/cosmos/ibc-go@v1.1.3
- github.com/davecgh/go-spew@v1.1.1
- github.com/desertbit/timer@v0.0.0-20180107155436-c41aec40b27f
- github.com/dvsekhvalnov/jose2go@v0.0.0-20200901110807-248326c1351b
- github.com/felixge/httpsnoop@v1.0.1
- github.com/fsnotify/fsnotify@v1.5.1
- github.com/go-kit/kit@v0.10.0
- github.com/go-logfmt/logfmt@v0.5.0
- github.com/godbus/dbus@v0.0.0-20190726142602-4481cbc300e2
- github.com/gogo/gateway@v1.1.0
- github.com/gogo/protobuf@v1.3.3 => github.com/regen-network/protobuf@v1.3.3-alpha.regen.1
- github.com/golang/protobuf@v1.5.2
- github.com/golang/snappy@v0.0.3
- github.com/google/btree@v1.0.0
- github.com/google/orderedcode@v0.0.1
- github.com/gorilla/handlers@v1.5.1
- github.com/gorilla/mux@v1.8.0
- github.com/gorilla/websocket@v1.4.2
- github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0
- github.com/grpc-ecosystem/grpc-gateway@v1.16.0
- github.com/gsterjov/go-libsecret@v0.0.0-20161001094733-a6f4afe4910c
- github.com/gtank/merlin@v0.1.1
- github.com/gtank/ristretto255@v0.1.2
- github.com/hashicorp/go-immutable-radix@v1.0.0
- github.com/hashicorp/golang-lru@v0.5.4
- github.com/hashicorp/hcl@v1.0.0
- github.com/hdevalence/ed25519consensus@v0.0.0-20210204194344-59a8610d2b87
- github.com/improbable-eng/grpc-web@v0.14.1
- github.com/klauspost/compress@v1.11.7
- github.com/lib/pq@v1.10.2
- github.com/libp2p/go-buffer-pool@v0.0.2
- github.com/magiconair/properties@v1.8.5
- github.com/mattn/go-isatty@v0.0.14
- github.com/matttproud/golang_protobuf_extensions@v1.0.1
- github.com/mimoo/StrobeGo@v0.0.0-20181016162300-f8f6d4d2b643
- github.com/minio/highwayhash@v1.0.1
- github.com/mitchellh/mapstructure@v1.4.2
- github.com/mtibben/percent@v0.2.1
- github.com/pelletier/go-toml@v1.9.4
- github.com/pkg/errors@v0.9.1
- github.com/pmezard/go-difflib@v1.0.0
- github.com/prometheus/client_golang@v1.11.0
- github.com/prometheus/client_model@v0.2.0
- github.com/prometheus/common@v0.29.0
- github.com/prometheus/procfs@v0.6.0
- github.com/rakyll/statik@v0.1.7
- github.com/rcrowley/go-metrics@v0.0.0-20200313005456-10cdbea86bc0
- github.com/regen-network/cosmos-proto@v0.3.1
- github.com/rs/cors@v1.7.0
- github.com/rs/zerolog@v1.23.0
- github.com/spf13/afero@v1.6.0
- github.com/spf13/cast@v1.3.1
- github.com/spf13/cobra@v1.2.1
- github.com/spf13/jwalterweatherman@v1.1.0
- github.com/spf13/pflag@v1.0.5
- github.com/spf13/viper@v1.8.1
- github.com/stretchr/testify@v1.7.0
- github.com/subosito/gotenv@v1.2.0
- github.com/syndtr/goleveldb@v1.0.1-0.20200815110645-5c35d600f0ca
- github.com/tendermint/btcd@v0.1.1
- github.com/tendermint/crypto@v0.0.0-20191022145703-50d29ede1e15
- github.com/tendermint/go-amino@v0.16.0
- github.com/tendermint/tendermint@v0.34.14
- github.com/tendermint/tm-db@v0.6.4
- golang.org/x/crypto@v0.0.0-20210817164053-32db794688a5
- golang.org/x/net@v0.0.0-20210903162142-ad29c8ab022f
- golang.org/x/sys@v0.0.0-20210903071746-97244b99971b
- golang.org/x/term@v0.0.0-20201126162022-7de9c90e9dd1
- golang.org/x/text@v0.3.6
- google.golang.org/genproto@v0.0.0-20210828152312-66f60bf46e71
- google.golang.org/grpc@v1.42.0 => google.golang.org/grpc@v1.33.2
- google.golang.org/protobuf@v1.27.1
- gopkg.in/ini.v1@v1.63.2
- gopkg.in/yaml.v2@v2.4.0
- gopkg.in/yaml.v3@v3.0.0-20210107192922-496545a6307b
- nhooyr.io/websocket@v1.8.6
cosmos_sdk_version: v0.44.5
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -15,8 +15,8 @@ WORKDIR /code
 COPY . /code/
 
 # See https://github.com/CosmWasm/wasmvm/releases
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v0.16.2/libwasmvm_muslc.a /lib/libwasmvm_muslc.a
-RUN sha256sum /lib/libwasmvm_muslc.a | grep 0e62296b9f24cf3a05f8513f99cee536c7087079855ea6ffb4f89b35eccdaa66
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v0.16.3/libwasmvm_muslc.a /lib/libwasmvm_muslc.a
+RUN sha256sum /lib/libwasmvm_muslc.a | grep 3fc6d5a239f3e97ac96c1a2df3006e4107ca461da4ca318bc71cfdc3e3593125
 
 # force it to use static lib (from above) not standard libgo_cosmwasm.so file
 RUN LEDGER_ENABLED=false BUILD_TAGS=muslc make build

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ go 1.17
 module github.com/terra-money/core
 
 require (
-	github.com/CosmWasm/wasmvm v0.16.2
+	github.com/CosmWasm/wasmvm v0.16.3
 	github.com/cosmos/cosmos-sdk v0.44.5
 	github.com/cosmos/ibc-go v1.1.5
 	github.com/gogo/protobuf v1.3.3

